test/e2e/framework/framework.go

```diff
@@ -234,6 +237,45 @@ func DetectGlobalnet() {
 	})
 }
 
+func fetchClusterIDs() {
+	for i := range KubeClients {
```

Comment:
nit: I'd restructure it so that this function receives the index and returns the cluster ID
Then when called it can be run in a loop from `BeforeSuite` and the code will be more straightforward and easier to read (also one indentation level less in the function).

We can do it in a follow up patch, though.